pallets/acurast/src/mock.rs

```diff
@@ -0,0 +1,225 @@
+        Acurast: crate::{Pallet, Call, Storage, Event<T>},
+    }
+);
+
+impl frame_system::Config for Test {
+    type BaseCallFilter = frame_support::traits::Everything;
+    type BlockWeights = BlockWeights;
+    type BlockLength = ();
+    type DbWeight = ();
+    type Origin = Origin;
+    type Index = u64;
+    type BlockNumber = u64;
+    type Call = Call;
+    type Hash = sp_core::H256;
+    type Hashing = ::sp_runtime::traits::BlakeTwo256;
+    type AccountId = u64;
+    type Lookup = IdentityLookup<Self::AccountId>;
+    type Header = Header;
+    type Event = Event;
+    type BlockHashCount = frame_support::traits::ConstU64<250>;
+    type Version = ();
+    type PalletInfo = PalletInfo;
+    type AccountData = ();
+    type OnNewAccount = ();
+    type OnKilledAccount = ();
+    type SystemWeightInfo = ();
+    type SS58Prefix = ();
+    type OnSetCode = ();
+    type MaxConsumers = frame_support::traits::ConstU32<16>;
+}
+
+frame_support::parameter_types! {
+    pub BlockWeights: frame_system::limits::BlockWeights = frame_system::limits::BlockWeights::simple_max(1024);
+    pub const MinimumPeriod: u64 = 6000;
+    pub AllowedRevocationListUpdate: Vec<AccountId> = vec![1];
+    pub static ExistentialDeposit: u64 = 0;
+}
+
+impl pallet_timestamp::Config for Test {
+    type Moment = u64;
+    type OnTimestampSet = ();
+    type MinimumPeriod = MinimumPeriod;
+    type WeightInfo = ();
+}
+
+impl crate::Config for Test {
+    type Event = Event;
+    type RegistrationExtra = ();
+    type FulfillmentRouter = Router;
+    type MaxAllowedSources = frame_support::traits::ConstU16<4>;
+    type AllowedRevocationListUpdate = AllowedRevocationListUpdate;
+}
+
+pub struct Router;
+
+impl crate::FulfillmentRouter<Test> for Router {
+    fn received_fulfillment(
+        _origin: frame_system::pallet_prelude::OriginFor<Test>,
+        _from: <Test as frame_system::Config>::AccountId,
+        _fulfillment: crate::Fulfillment,
+        _registration: crate::JobRegistration<
+            <Test as frame_system::Config>::AccountId,
+            <Test as crate::Config>::RegistrationExtra,
+        >,
+        _requester: <<Test as frame_system::Config>::Lookup as sp_runtime::traits::StaticLookup>::Target,
+    ) -> frame_support::pallet_prelude::DispatchResultWithPostInfo {
+        Ok(().into())
+    }
+}
+
+pub struct ExtBuilder;
+
+impl ExtBuilder {
+    pub fn build(self) -> sp_io::TestExternalities {
+        let t = frame_system::GenesisConfig::default()
+            .build_storage::<Test>()
+            .unwrap();
+
+        let mut ext = sp_io::TestExternalities::new(t);
+        ext.execute_with(|| System::set_block_number(1));
+        ext
+    }
+}
+
+impl Default for ExtBuilder {
+    fn default() -> Self {
+        Self {}
+    }
+}
+
+pub fn events() -> Vec<Event> {
+    let evt = System::events()
+        .into_iter()
+        .map(|evt| evt.event)
+        .collect::<Vec<_>>();
+
+    System::reset_events();
```

Comment:
this resets the events when retrieving them, so if you forget (or decide) to not check for them, a subsequent test asserting events will break.

Reply:
@gitsimon I do not think this is the case between tests invocations, you could say that within the same test, but that is not really an issue.